src/storage/heap_file.c

```diff
@@ -21218,6 +21231,16 @@ heap_delete_relocation (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEXT * contex
 
       HEAP_PERF_TRACK_EXECUTE (thread_p, context);
 
+      /*
+       * When update_old_forward is set it signifies that more than one page will be modified,
+       * and so the changes need to be marked as atomic for replication
+       */
+      if (update_old_forward)
+	{
+	  log_append_empty_record (thread_p, LOG_START_ATOMIC_REPL, NULL);
+	  atomic_replication_flag = true;
+	}
```

Comment:
It is the other way around :). When `update_old_forward`, only the record in forward page is updated. When `update_old_forward` is false, it means the record is either moved back home, moved in another page, or converted into a big record; in all cases two or more pages are modified and atomic replication is required.

```suggestion
      /*
       * When update_old_forward is true it signifies that only the forward page will be modified,
       * In all other cases two or more pages are changed and need to be marked as atomic for replication
       */
      if (!update_old_forward)
	{
	  log_append_empty_record (thread_p, LOG_START_ATOMIC_REPL, NULL);
	  atomic_replication_flag = true;
	}
```